homeassistant/components/roomba/sensor.py

```diff
@@ -42,11 +43,16 @@ def unit_of_measurement(self):
         """Return the unit_of_measurement of the device."""
         return UNIT_PERCENTAGE
 
+    @property
+    def icon(self):
```

Comment:
We don't add custom icon to sensor entities with device class set. That should be handled by the frontend.

Reply:
That totally makes sense. But to achieve this, the frontend will need to know if the battery is charging. Maybe we need to add a property to sensor? This is gonna be an architecture issue.

Reply:
We added a binary sensor device class for charging recently.

https://github.com/home-assistant/architecture/issues/359

So with that entity the frontend should have the possibility to know if a device is charging or not, if the integration supports the device registry.

I'm not sure what the frontend status is.

Reply:
@MartinHjelmare Thanks. I'll take a look.

Reply:
The basic idea is to make a binary sensor with device class `battery_charging`.  I haven't finished the frontend work to make the frontend aware of these yet.  The old helper method you used here is currently the functional method and should no longer be needed once I do the frontend work.